ui/src/app/base/selectors/machine/machine.js

```diff
@@ -69,12 +69,16 @@ machine.getBySystemId = (state, id) =>
 machine.errors = state => state.machine.errors;
 
 /**
- * Gets the search terms.
+ * Gets the search terms and selected machines.
  * @param {Object} state - The redux state.
- * @param {Array} terms - The search terms.
- * @returns {Array} The search terms.
+ * @param {String} terms - The search string.
+ * @param {Array} selectedIDs - The selected machine ids.
+ * @returns {Array} The search terms and selected machines.
  */
-machine._getTerms = (state, terms) => terms;
+machine._getSearchParams = (state, terms, selectedIDs) => ({
```

Comment:
This is interesting way to curry arguments into selectors. You may prefer how we do it on the jaas-dashboard https://github.com/canonical-web-and-design/jaas-dashboard/blob/master/src/app/selectors.js#L448

Reply:
I decided against this approach as it makes using the selector more complicated (looks like you wrap the selector creator in `useMemo` before you use `useSelector` on the resulting function.

Reply:
Sure no problem, maybe we can convert the dashboard to use this approach instead. Something to investigate.